roles/kubernetes-apps/meta/main.yml

```diff
@@ -57,3 +57,13 @@ dependencies:
       - inventory_hostname == groups['kube-master'][0]
     tags:
       - oci
+
+  - role: kubernetes-apps/cloud_controller/openstack
```

Comment:
In my experiments I needed to deploy the controller manager *before* deploying the network plugin (Weave in my case). Otherwise the Weave pods wouldn't start due to the Kubernetes nodes not having been assigned an internal IP yet (which is something the OpenStack controller manager provides).

In your current code it's being deployed after the network plugin.

Reply:
I don't think the OpenStack controller manager assigns addresses to the pods. Maybe the problem is with node initialization?

Reply:
I'll test it again to make sure, but in my experience if kubelet is started with `--cloud-provider=external` Weave has trouble starting up which makes kubespray fail. Deploying the CCM before the CNI made it work.

Maybe since the `--cloud-provider=external` config for kubelet is missing in this PR you might not have encountered this error? I use `kubelet_custom_flags: --cloud-provider=external` but you could also modify the InitConfiguration of `kubeadm.conf` as you already mentioned in your previous comment.

Reply:
I just updated the PR with a great deal of modifications. Can you check if it works in your configuration? Thanks!

Reply:
Thanks a lot! I'll try it out on our setup on Monday.